django-stubs/db/models/manager.pyi

```diff
@@ -108,19 +108,28 @@ class Manager(BaseManager[_T]):
     def values(self, *fields: str | Combinable, **expressions: Any) -> QuerySet[_T, dict[str, Any]]: ...
     # The type of values_list may be overridden to be more specific in the mypy plugin, depending on the fields param
     def values_list(self, *fields: str | Combinable, flat: bool = ..., named: bool = ...) -> QuerySet[_T, Any]: ...
-    def dates(self, field_name: str, kind: str, order: str = ...) -> QuerySet[_T, datetime.date]: ...
+    def dates(
+        self,
+        field_name: str,
+        kind: Literal["year", "month", "week", "day"],
+        order: Literal["ASC", "DESC"] = "ASC",
+    ) -> QuerySet[_T, datetime.date]: ...
     def datetimes(
-        self, field_name: str, kind: str, order: str = ..., tzinfo: datetime.tzinfo | None = ...
+        self,
+        field_name: str,
+        kind: Literal["year", "month", "week", "day", "hour", "minute", "second"],
+        order: Literal["ASC", "DESC"] = "ASC",
+        tzinfo: datetime.tzinfo | None = None,
     ) -> QuerySet[_T, datetime.datetime]: ...
     def none(self) -> QuerySet[_T]: ...
     def filter(self, *args: Any, **kwargs: Any) -> QuerySet[_T]: ...
     def exclude(self, *args: Any, **kwargs: Any) -> QuerySet[_T]: ...
     def complex_filter(self, filter_obj: Any) -> QuerySet[_T]: ...
     def count(self) -> int: ...
     async def acount(self) -> int: ...
-    def union(self, *other_qs: Any, all: bool = ...) -> QuerySet[_T]: ...
-    def intersection(self, *other_qs: Any) -> QuerySet[_T]: ...
-    def difference(self, *other_qs: Any) -> QuerySet[_T]: ...
+    def union(self, *other_qs: QuerySet[Model], all: bool = ...) -> QuerySet[_T]: ...
```

Comment:
```suggestion
    def union(self, *other_qs: QuerySet[_T], all: bool = ...) -> QuerySet[_T]: ...
```

I don't think that we can `.union` two different models, can't we?

Reply:
We can! (But I would not recommend it because of how disturbing the result is) 

> union(), intersection(), and difference() return model instances of the type of the first QuerySet even if the arguments are QuerySets of other models. Passing different models works as long as the SELECT list is the same in all QuerySets (at least the types, the names don’t matter as long as the types are in the same order). In such cases, you must use the column names from the first QuerySet in QuerySet methods applied to the resulting QuerySet. 

ref: https://docs.djangoproject.com/en/5.2/ref/models/querysets/#union